cartridge/roles/expirationd.lua

```diff
@@ -2,15 +2,31 @@ local expirationd = require("expirationd")
 local role_name = "expirationd"
 local started = require("cartridge.vars").new(role_name)
 
+local function dumb_fn() end
+
 local function load_function(func_name)
     if func_name == nil or type(func_name) ~= 'string' then
         return nil
     end
 
     local func = rawget(_G, func_name)
-    if func == nil or type(func) ~= 'function' then
+    if func == nil then
+        if type(box.cfg) == 'function' then
+            return dumb_fn
```

Comment:
It seems to me that it is better to return `nil` in this case and signal about a problem. It is better to get an error at the time of checking the config than at runtime.

Or the configuration has not yet been completed at the `validate_config` step?

Reply:
Yes, here is a problem with restarting nodes of cartridge application.
After restart `validate_config` sometimes is run before `box.cfg` call and it leads to error like `Please call box.cfg first` and at this moment we are not able to check real availability of functions in box.func. If we return nil, we fail configration but it is not an error actually, because we cannot do any checks. Thus we decided to return dumb_fn to avoid misconfiguration at this stage and in hope that apply_config will do real check.

Reply:
Thank you for the explanation.

This is not very good, because in `apply_config` we need to have a validated (and valid) configuration. But it seems that we could not to do anything here:

https://www.tarantool.io/en/doc/latest/book/cartridge/cartridge_dev/#role-s-life-cycle-and-the-order-of-function-execution
`The validate_config() function, first, before the automatic box.cfg() call (database initialization), then – upon every configuration update.`